Chapters/6_Qualified_identifiers.md

```diff
@@ -18,13 +18,29 @@ The following *context qualifiers* are available:
 A "line" in the context of a file content refers to a sequence of characters that ends with a line break. This line can contain text, code, or any other form of data. In this specification, the line break is the ASCII LF character.
 The "lines" qualifier allows to designate a line range inside a content.
 The range can be a single line number, or a pair of line numbers separated by the ASCII `-` character.
+Line numbers start from 1.
 
 For example, [`swh:1:cnt:4d99d2d18326621ccdd70f5ea66c2e2ac236ad8b;lines=9-15`](https://archive.softwareheritage.org/swh:1:cnt:4d99d2d18326621ccdd70f5ea66c2e2ac236ad8b;lines=9-15)
 designates the function `generate_intput_stream` that is found at lines 9 to 15 of the *content* with core SWHID `swh:1:cnt:4d99d2d18326621ccdd70f5ea66c2e2ac236ad8b`.
 
 Notice that the notion of "line number" is not always meaningful: the content
 may be a binary file, or a file that uses non standard line termination character(s).
 
+### 6.1.2 Bytes qualifier
+
+To overcome the limitations of the lines qualifier, the bytes qualifier allows
+to designate a byte range inside a content.  The range can be a single byte number, or a pair of byte numbers separated by `-`.
+Byte numbers start from 0.
+If the range is a single byte number, it designates the byte at that specific position.
+
+For example, `swh:1:cnt:4d99d2d18326621ccdd70f5ea66c2e2ac236ad8b;bytes=154-315`
+designates the same function `generate_intput_stream` as in the example above, but
+does not rely on any convention about line numbers.
+
+### 6.1.3 Bytes and line qualifiers are mutually exclusive
+
+As clearly defined in the grammar, bytes and line qualifiers are mutually exclusive, so only one of them can be used in a valid SWHID. A conformant implementation MAY accept a SWHID that violates this constraint, by ingoring the `lines` qualifier when a `bytes` qualifier is present.
```

Comment:
Do we really want to empower/suggest that conformant implementations should override lines with bytes?
I'm not strongly opposed to this, but it seems to add complexity for little/unclear gain.

Proposed alternative wording, which also remove the redundant reference to the grammar and is overall much simpler:
```
"`bytes` and `lines` qualifiers are mutually exclusive: a valid SWHID MUST not contain both."
```

Reply:
My take is that there will be malformed SWHIDs no matter what we mandate, so it will help stating how implementations should deal with them, if they decide not to reject them.

Reply:
Incorporated in 1649c2910ff819d794440683fe483494cd7bc00d